client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamImpl.java

```diff
@@ -212,18 +203,19 @@ private void verifyIsAtCorrectOffset(WireCommands.SegmentRead segmentRead) {
      */
     private void issueRequestIfNeeded() {
         //compute read length based on current offset up to which the events are read.
-        int updatedReadLength = computeReadLength(offset + buffer.dataAvailable(), readLength);
-        if (!receivedEndOfSegment && !receivedTruncated && updatedReadLength > 0 && buffer.capacityAvailable() >= updatedReadLength && outstandingRequest == null) {
+        int updatedReadLength = computeReadLength(offset + buffer.dataAvailable());
+        if (!receivedEndOfSegment && !receivedTruncated && updatedReadLength > 0 && outstandingRequest == null) {
             outstandingRequest = asyncInput.read(offset + buffer.dataAvailable(), updatedReadLength);
         }
     }
 
     /**
      * Compute the read length based on the current fetch offset and the configured end offset.
      */
-    private int computeReadLength(long currentFetchOffset, int currentReadLength) {
+    private int computeReadLength(long currentFetchOffset) {
         Preconditions.checkState(endOffset >= currentFetchOffset,
                 "Current offset up to to which events are fetched should be less than the configured end offset");
+        int currentReadLength = Math.max(minReadLength, buffer.capacityAvailable());
```

Comment:
Can it so happen that the `currentReadLength` is greater than `buffer.capacityAvailable()` and less than `(endOffset-currentFetchOffset)` ?

Reply:
Sure suppose the buffer has 1kb available, and the endOffset is maxLong. Then it will fall back on the minReadLength which is 256kb.